detectors/node/opentelemetry-resource-detector-docker/test/DockerCGroupv1Detector.test.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as sinon from 'sinon';
+import * as assert from 'assert';
+import { Resource } from '@opentelemetry/resources';
+import { dockerCGroupV1Detector } from '../src';
+import {
+  assertContainerResource,
+  assertEmptyResource,
+} from '@opentelemetry/contrib-test-utils';
+
+import { DockerCGroupV1Detector } from '../src';
+
+describe('dockerCGroupV1Detector', () => {
+  let readStub;
+  const correctCgroupData =
+    'abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklm';
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('Supported docker - Container ID ', () => {
+    it('should return a container ID for supported docker - cgroup v1', async () => {
+      const resource: Resource = await dockerCGroupV1Detector.detect();
+      assert.ok(resource);
+    });
+
+    it('should return a resource with container ID with a valid container ID present', async () => {
+      readStub = sinon
+        .stub(DockerCGroupV1Detector, 'readFileAsync' as any)
+        .resolves(correctCgroupData);
+
+      const resource: Resource = await dockerCGroupV1Detector.detect();
+
+      sinon.assert.calledOnce(readStub);
+
+      assert.ok(resource);
+      assertContainerResource(resource, {
+        id: 'bcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklm',
+      });
+    });
+
+    it('should return a resource with clusterName attribute when cgroup file does not contain valid Container ID', async () => {
+      readStub = sinon
+        .stub(DockerCGroupV1Detector, 'readFileAsync' as any)
+        .onSecondCall()
+        .resolves('');
+
+      const resource: Resource = await dockerCGroupV1Detector.detect();
+
+      sinon.assert.calledOnce(readStub);
+
+      assert.ok(resource);
+    });
+
+    it('should return an empty resource when containerId is not valid', async () => {
+      const errorMsg = {
+        fileNotFoundError: new Error('cannot file in the path}'),
+      };
+
+      readStub = sinon
+        .stub(DockerCGroupV1Detector, 'readFileAsync' as any)
+        .onSecondCall()
+        .rejects(errorMsg);
```

Comment:
Maybe I misunderstand you, but the test uses:
```js
      const errorMsg = {
        fileNotFoundError: new Error('cannot file in the path}'),
      };
```

And it will not return an error, but an object with one key named `fileNotFoundError` which is an `Error`.
Is it consistent with what `fs` returns on `err`?
I would expect it to be something like:
```js
.rejects(new Error('foo bar'));
```